dev/tasks/jars/Dockerfile

```diff
@@ -0,0 +1,124 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+FROM quay.io/pypa/manylinux2014_x86_64:latest
```

Comment:
Thanks @anthonylouisbsb for working on this!

I wonder, could we reuse the [existing manylinux dockerfile](https://github.com/apache/arrow/blob/master/ci/docker/python-wheel-manylinux-201x.dockerfile) which uses vcpkg to build the dependencies?

Reply:
@kszucs Thanks for your help! I did not notice that dockerfile.

I think I can use the [docker image from Arrow's Docker Hub repository](https://hub.docker.com/layers/apache/arrow-dev/amd64-python-3.7-wheel-manylinux-2014/images/sha256-e3691bcc875e7ce7ec633a0d2c9d8346c65f0f3bfabfd9f45ec1d2cbc32b3138?context=explore), but I need to update the Dockerfile that you sent in the message to import **LLVM** and **CLANG** using vcpkg and generate a new docker image in the hub.

Do you think is better to do it in this Pull Request or should I need to open a new one to apply that changes?

Reply:
I'd suggest to create a child image which depends on the manylinux2014 image and install the additional packages required for gandiva there. This way we wouldn't need to check in and maintain that many scripts.

Something like the following should be enough:

#### ci/scripts/java_bundled_jar.sh

This file could be a customized version of `ci/scripts/python_wheel_manylinux_build.sh` or the [build-linux-cpp.sh]( https://github.com/apache/arrow/pull/10300/files#diff-eb09a993bfee5b11462aecb3e85df34161bbf5b82a792fceacf545adba5e6b8f) without multibuild.

#### ci/docker/java-bundled-jar.dockerfile:

```dockerfile
ARG base
FROM ${base}

RUN vcpkg install --clean-after-build <additional-dependencies>

ARG java=1.8.0
RUN yum install -y xz java-$java-openjdk-devel ccache flex wget && yum clean all
ENV JAVA_HOME=/usr/lib/jvm/java-1.8.0-openjdk/
```

#### docker-compose.yml

```yml
x-hierarchy:
  # ...
  python-wheel-manylinux-2014:
    - java-bundled-jar:

services:
  # ...
  java-bundled-jar:
    image: ${REPO}:${ARCH}-java-bundled-jar-vcpkg-${VCPKG}
    build:
      args:
        base: ${REPO}:${ARCH}-python-${PYTHON}-wheel-manylinux-2014-vcpkg-${VCPKG}
        java: 1.8.0
      context: .
      dockerfile: ci/docker/java-bundled-jar.dockerfile
      cache_from:
        - ${REPO}:${ARCH}-java-bundled-jar-vcpkg-${VCPKG}
    environment:
      <<: *ccache
    volumes:
      - .:/arrow:delegated
      - ${DOCKER_VOLUME_PREFIX}python-wheel-manylinux2014-ccache:/ccache:delegated
    command: /arrow/ci/scripts/java_bundled_jar.sh
```

Then you can use archery to build and run this docker setup:

```console
archery docker build java-bundled-jar  # just builds the images
archery docker run java-bundled-jar  # builds and executes the image
```

How does it sound to you?

Reply:
It sounds good, I think it is a good idea. I will apply the changes you suggested.